pytorch_lightning/trainer/optimizers.py

```diff
@@ -53,15 +58,25 @@ def init_optimizers(self, model: LightningModule) -> Tuple[List, List, List]:
         # multiple dictionaries
         elif isinstance(optim_conf, (list, tuple)) and all(isinstance(d, dict) for d in optim_conf):
             optimizers = [opt_dict["optimizer"] for opt_dict in optim_conf]
-            lr_schedulers = [opt_dict["lr_scheduler"] for opt_dict in optim_conf if "lr_scheduler" in opt_dict]
+            scheduler_dict = (
+                lambda scheduler, opt_idx: dict(scheduler, opt_idx=opt_idx)
+                if isinstance(scheduler, dict)
+                else dict({'scheduler': scheduler, 'opt_idx': opt_idx})
+            )
+
+            lr_schedulers = [
+                scheduler_dict(opt_dict["lr_scheduler"], opt_idx)
+                for opt_idx, opt_dict in enumerate(optim_conf)
+                if "lr_scheduler" in opt_dict
+            ]
```

Comment:
I think you can just index `self.trainer.lr_schedulers` in `update_learning_rates` using the opt_indices. Does it need a separate opt_idx key for any other purpose?

Reply:
`opt_idx` is required because not all optimizers will have LR schedulers. For instance, if two optimizers are passed and only the second one has an LR scheduler, `self.trainer.lr_schedulers[0]` will refer to the LR scheduler for optimizer at index 1.

Reply:
makes sense :+1: